images/git-mirror/code/repository.go

```diff
@@ -57,6 +62,40 @@ func (r *repository) fetch(reason string) {
 	}
 
 	log.WithFields(log.Fields{"Reason": reason, "Repo": r.name}).Debug("Fetched")
+	r.parsePackedRefs()
+}
+
+func (r *repository) parsePackedRefs() {
+	packedRefPath := strings.Join([]string{r.targetDir, "packed-refs"}, "/")
+
+	var advance int
+	var token []byte
+	if data, err := ioutil.ReadFile(packedRefPath); err == nil {
+		for {
+			advance, token, err = bufio.ScanLines(data, false)
+			if advance == 0 {
+				break
+			}
+			data = data[advance:]
+			if err != nil {
+				log.Warn("Error scanning packed-refs: %+v", err)
+				break
+			}
+			line := string(token)
+			if line[0:1] == "#" {
+				continue
+			}
+			r.refs[strings.Trim(line[40:], " ")] = line[:40]
```

Comment:
Need to build a new array and overwrite `r.refs` instead of modifying the existing array